lib/eslint.js

```diff
@@ -819,11 +820,14 @@ module.exports = (function() {
     api.report = function(ruleId, severity, node, location, message, opts, fix) {
 
         if (typeof location === "string") {
+            assert.ok(node, "Node must be provided when reporting error if location is not provided");
```

Comment:
We should check that it's an object. If it's a string, for instance, it should also fail.